src/main/java/fi/metatavu/metamind/server/keycloak/AuthenticationController.java

```diff
@@ -0,0 +1,408 @@
+    Keycloak keycloak = getAdminClient();
+    ClientRepresentation client = getClient(keycloak);
+    if (client == null) {
+      throw new AuthorizationException();
+    }
+    RealmResource realm = keycloak.realm(realmName);
+    ScopePermissionsResource scopeResource = realm.clients().get(client.getId()).authorization().permissions().scope();
+    ScopePermissionRepresentation existingPermission = scopeResource.findByName(name);
+
+    ScopePermissionRepresentation representation = new ScopePermissionRepresentation();
+    representation.setResources(Collections.singleton(resourceId.toString()));
+    representation.setDecisionStrategy(decisionStrategy);
+    representation.setName(name);
+    representation.setScopes(scopes.stream().map(AuthorizationScope::getName).collect(Collectors.toSet()));
+    representation.setPolicies(policyIds.stream().map(UUID::toString).collect(Collectors.toSet()));
+
+    String result = representation.getName();
+    Response response = scopeResource.create(representation);
+    try {
+      if (existingPermission == null) {
+        int status = response.getStatus();
+        if (status != 201) {
+          String message = "Unknown error";
+          try {
+            message = IOUtils.toString((InputStream) response.getEntity(), StandardCharsets.UTF_8);
+          } catch (IOException e) {
+            logger.warn("Failed read error message", e);
+          }
+
+          logger.warn("Failed to create scope permission for resource {} with message {}", resourceId, message);
+        }
+        
+      } else {
+        realm.clients().get(client.getId()).authorization().permissions().scope().findById(existingPermission.getId()).update(representation);
+      }
+      
+    } finally {
+      response.close();
+    }
+    
+    return result;
+  }
+  
+
+  
+  /**
+   * Updates users and user policies into Keycloak
+   * 
+   * @param keycloak admin client
+   * @param realmName realm name
+   * @param clientId client id
+   * @param userMap users names
+   */
+  public List<UUID> updatePermissionUsers(String realmName) throws AuthorizationException {
+    List<UUID> result = new ArrayList();
+    Keycloak keycloak = getAdminClient();
+    RealmResource realm = keycloak.realm(realmName);
+    UsersResource users = realm.users();
+    ClientRepresentation client = getClient(keycloak);
+    if (client == null) {
+      throw new AuthorizationException();
+    }
+    UserPoliciesResource userPolicies = realm.clients().get(client.getId()).authorization().policies().user();
+    
+    List<UserRepresentation> existingUsers = users.list().stream().collect(Collectors.toList());
+    
+    for (UserRepresentation userRep : existingUsers) {
+      UUID userId = UUID.fromString(userRep.getId());
+      String userName = userRep.getUsername();
+      
+      result.add(userId);
+
+      UserPolicyRepresentation policyRepresentation = userPolicies.findByName("user-" + userName);
```

Comment:
Replace "user-" with String.format("user-%s", userName) and move to local variable for future use